### PR TITLE
Revise events spreaded with send_message

### PR DESCRIPTION
**Pull request by rosik**
- Avoid event duplication due to a bug.

### CODE REVIEW

File: test/test_false_rumors.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    # Ack timeout shouldn't trigger failure detection
+    # because inderect pings still work
+
+    servers[13301].conn.eval('''
+        local opts = require('membership.options')
+        opts.ACK_TIMEOUT_SECONDS = 0
+    ''')
+
+    time.sleep(2)
+    check_rumors(servers[13301], {})
```
Comment (olegrok) [inline test/test_false_rumors.py:56]: wait_for?

File: membership.lua [hunk 2]
```diff
@@ -126,37 +126,33 @@ local function send_message(uri, msg_type, msg_data)
...
+            status = myself.status,
             incarnation = myself.incarnation,
             payload = myself.payload,
             ttl = 1,
         }
-    end
-    if extra_event ~= nil then
-        table.insert(events_to_send, events.pack(extra_event))
-        msg_size = msg_size + events.estimate_msgpacked_size(extra_event)
+        -- Don't tell anyone about dead myself
```
Comment (olegrok) [inline membership.lua:151]: It would be great to describe a reason here.

File: membership.lua [hunk 3]
```diff
@@ -126,37 +126,33 @@ local function send_message(uri, msg_type, msg_data)
...
+        if extra_event.status == opts.ALIVE then
+            table.insert(events_to_send, events.pack(extra_event))
+            msg_size = msg_size + events.estimate_msgpacked_size(extra_event)
```
Comment (olegrok) [inline membership.lua:154]: Side question.

Why https://github.com/tarantool/membership/blob/e479849c424ade4747aedce99203e3d4a0115cc6/membership/events.lua#L53 unpacks event and doesn't call `#msgpack.encode(event)` directly?

File: test/test_false_rumors.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        end
+
+        _G._protection_fiber = fiber.create(function()
+            local cond = membership.subscribe()
+            while true do
+                cond:wait()
+                fiber.testcancel()
+                collect_rumors()
+            end
+        end)
```
Comment (olegrok) [inline test/test_false_rumors.py:36]: It's worth to describe what this monkeypatch does.

File: membership.lua (hunk 3 above)
Comment (rosik) [reply membership.lua:154]: `event` here is a key-value table. It should be `#msgpack.encode(pack(event))`, but `pack` implies other logick to count ttl.

File: test/test_false_rumors.py (hunk 1 above)
Comment (rosik) [reply test/test_false_rumors.py:56]: Nope. Rumors are always appended in this test. In other words, this check means that there were no rumors in past 2 sec.

File: membership.lua (hunk 3 above)
Comment (olegrok) [reply membership.lua:154]: Ok. Thanks for explanation.

File: membership.lua (hunk 2 above)
Comment (rosik) [reply membership.lua]: I've tried to explain it, but couldn't. Moreover, your question pointed me to a thought that this change is incorrect. So I've deleted it at all.

File: membership.lua [hunk 5]
```diff
@@ -126,42 +126,38 @@ local function send_message(uri, msg_type, msg_data)
...
             ttl = 1,
         }
-    end
-    if extra_event ~= nil then
         table.insert(events_to_send, events.pack(extra_event))
         msg_size = msg_size + events.estimate_msgpacked_size(extra_event)
         events_to_send[advertise_uri] = true
     end
 
     for _, event in events.pairs() do
```
Comment (olegrok) [inline membership.lua:159]: nit: seems you can to change this line to `for uri, event in events.pairs() do`.

Comment (rosik) [reply membership.lua]: Done

File: test/test_false_rumors.py (hunk 4 above)
Comment (rosik) [reply test/test_false_rumors.py]: Done

**Review by olegrok — APPROVED** [review]

**Pull request merged**